sdk/extensions/chronopost/tests/chronopost/tracking.py

```diff
@@ -0,0 +1,267 @@
+import unittest
+from unittest.mock import patch, ANY
+from tests.chronopost.fixture import gateway
+
+import karrio
+import karrio.lib as lib
+import karrio.core.models as models
+
+
+class TestchronopostTracking(unittest.TestCase):
+    def setUp(self):
+        self.maxDiff = None
+        self.TrackingRequest = models.TrackingRequest(**TrackingPayload)
+
+    def test_create_tracking_request(self):
+        request = gateway.mapper.create_tracking_request(self.TrackingRequest)
+        self.assertEqual(request.serialize(), [TrackingRequest])
+
+    def test_get_tracking(self):
+        with patch("karrio.mappers.chronopost.proxy.lib.request") as mock:
+            mock.return_value = "<a></a>"
+            karrio.Tracking.fetch(self.TrackingRequest).from_(gateway)
+
+            self.assertEqual(
+                mock.call_args[1]["url"],
+                f"{gateway.settings.server_url}/tracking-cxf/TrackingServiceWS",
+            )
+
+    def test_parse_tracking_response(self):
+        with patch("karrio.mappers.chronopost.proxy.lib.request") as mock:
+            mock.return_value = TrackingResponse
+            parsed_response = (
+                karrio.Tracking.fetch(self.TrackingRequest).from_(gateway).parse()
+            )
+            print(lib.to_dict(parsed_response))
+            self.assertListEqual(lib.to_dict(parsed_response), ParsedTrackingResponse)
+
+    # def test_parse_error_response(self):
+    #     with patch("karrio.mappers.chronopost.proxy.lib.request") as mock:
+    #         mock.return_value = ErrorResponse
+    #         parsed_response = (
+    #             karrio.Tracking.fetch(self.TrackingRequest).from_(gateway).parse()
+    #         )
+
+    #         self.assertListEqual(lib.to_dict(parsed_response), ParsedErrorResponse)
```

Comment:
I would recommend adding a the test for tracking error parsing.

Reply:
Okay, I wanted to leave parsing errors for last because it's kind of weird. All chronopost responses are in a `<return></return>`  tag and in that tag you always find `errorCode` which is 0 if everything is fine and `errorMessage` that is sent when an error occures. I couldn't figure out how to parse this.